File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    alp = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"]
    num = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"]
    
    result = ''
    temp = ''
    
    for val in s:
        if val in num:
            result += val
        else:
            temp += val
            if temp in alp:
                result += str(alp.index(temp))
                temp = ''
                
    return int(result)

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
function solution(s) {
    const alp = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"];
    const num = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"];
    
    let result = '';
    let temp = '';
    
    for (let i = 0; i < s.length; i++) {
        if (num.includes(s[i])) {
            result += s[i];
        } else {
            temp += s[i];
            if (alp.includes(temp)) {
                result += String(alp.indexOf(temp));
                temp = '';
            }
        }
    }
    return Number(result);
}